xds/internal/balancer/wrrlocality/balancer_test.go

```diff
@@ -119,3 +130,135 @@ func (s) TestParseConfig(t *testing.T) {
 		})
 	}
 }
+
+func setup(t *testing.T) (*wrrLocalityBalancer, func()) {
+	t.Helper()
+	builder := balancer.Get(Name)
+	if builder == nil {
+		t.Fatalf("balancer.Get(%q) returned nil", Name)
+	}
+	tcc := testutils.NewTestClientConn(t)
+	wrrL := builder.Build(tcc, balancer.BuildOptions{})
+	return wrrL.(*wrrLocalityBalancer), wrrL.Close
+}
+
+// TestUpdateClientConnState tests the UpdateClientConnState method of the
+// wrr_locality_experimental balancer. This UpdateClientConn operation should
+// take the localities and their weights in the addresses passed in, alongside
+// the endpoint picking policy defined in the Balancer Config specified and
+// construct a weighted target configuration corresponding to these inputs.
+func (s) TestUpdateClientConnState(t *testing.T) {
+	cfgCh := testutils.NewChannel()
+	oldWeightedTargetName := weightedTargetName
+	defer func() {
+		weightedTargetName = oldWeightedTargetName
+	}()
+	// Overwrite the weighted target name to have wrrLocalityBalancer to pull
+	// the mock balancer defined below from the balancer registry to be it's
+	// child.
+	weightedTargetName = "mock_weighted_target"
+	stub.Register("mock_weighted_target", stub.BalancerFuncs{
+		ParseConfig: func(c json.RawMessage) (serviceconfig.LoadBalancingConfig, error) {
+			var cfg weightedtarget.LBConfig
+			if err := json.Unmarshal(c, &cfg); err != nil {
+				return nil, err
+			}
+			return &cfg, nil
+		},
+		UpdateClientConnState: func(bd *stub.BalancerData, ccs balancer.ClientConnState) error {
+			wtCfg, ok := ccs.BalancerConfig.(*weightedtarget.LBConfig)
+			if !ok {
+				return errors.New("child received config that was not a weighted target config")
+			}
+			defer cfgCh.Send(wtCfg)
+			return nil
+		},
+	})
+
+	wrrL, close := setup(t)
+	defer close()
+
+	// Create the addresses with two localities with certain locality weights.
+	// This represents what the addresses the wrr_locality balancer will receive
+	// in UpdateClientConnState.
```

Comment:
Fix grammar.

Reply:
Switched "what the addresses" to "what addresses".